sharing/handler.go

```diff
@@ -406,24 +407,41 @@ func (handler *Handler) handleHostingService(ctx context.Context, w http.Respons
 		}
 	}()
 
-	// e.g. http://mydomain.com/folder2/index.html with root="bucket1/folder1"
-	rootPath := strings.SplitN(root, "/", 2) // e.g. rootPath=[bucket1, folder1]
+	// e.g. http://mydomain.com/folder2/index.html with "storj_root:bucket1/folder1/"
+	rootPath := strings.SplitN(root, "/", 2) // e.g. rootPath=[bucket1, folder1/]
 	bucket := rootPath[0]
-	pt := strings.TrimPrefix(r.URL.Path, "/") // e.g. path = "folder2/index.html
+	pt := strings.TrimPrefix(r.URL.Path, "/") // e.g. pt = "folder2/index.html
 	if len(rootPath) == 2 {
-		pathPrefix := rootPath[1]  // e.g. pathPrefix = "folder1"
-		pt = pathPrefix + "/" + pt // e.g. path="folder1/folder2/index.html"
+		pathPrefix := rootPath[1] // e.g. pathPrefix = "folder1/"
+		pt = pathPrefix + pt      // e.g. "folder1/folder2/index.html"
 	}
-
-	if pt == "" {
-		pt = "index.html"
-	}
-
 	o, err := project.StatObject(ctx, bucket, pt)
-	if err != nil {
+	if err != nil && !strings.Contains(err.Error(), "object not found") {
```

Comment:
Use `errors.Is` for checking error conditions.

Reply:
I'm having trouble with `errors.Is()` 
The error changes depending on the path requested and looks like `uplink: object not found ("<path>")` 
Is there still a way to use `errors.Is()` given this issue?

Reply:
Yes, `errors.Is` should still work.

Reply:
Do you know what the target error should be? I even tried this but still no match (even though it looks like a match when printed):
```
func errwrapf(format string, err error, args ...interface{}) error {
	pe := errs.Class("uplink")
	var all []interface{}
	all = append(all, err)
	all = append(all, args...)
	return pe.Wrap(fmt.Errorf(format, all...))
}
ErrObjectNotFound := errors.New("object not found")
x :=errwrapf("%w (%q)", ErrObjectNotFound, pt)

```

Reply:
Use the errors defined in uplink package. This means `errors.Is(err, uplink.ErrObjectNotFound)`.

Reply:
Ok that seems to work now, Idk what I did earlier that it wasn't working with that uplink.ErrObjectNotFound. Thanks Egon

Reply:
done

Reply:
You can read more here https://blog.golang.org/go1.13-errors.

But in general `errors.Is` does not use the message of an error to compare them. So if you "redefine" an error you cannot use to compare them that way. (There are exceptions to this, but this means the specific error type needs to support it.)